pkg/backend/snapshot.go

```diff
@@ -60,13 +62,19 @@ type SnapshotManager struct {
 	doVerify         bool                     // If true, verify the snapshot before persisting it
 	plugins          []workspace.PluginInfo   // The list of plugins loaded by the plan, to be saved in the manifest
 	mutationRequests chan func()              // The queue of mutation requests, to be retired serially by the manager
+	hasElidedWrites  bool                     // True if we have elided writes since the last actual write.
 }
 
 var _ engine.SnapshotManager = (*SnapshotManager)(nil)
 
 func (sm *SnapshotManager) Close() error {
+	var err error
+	if sm.hasElidedWrites {
```

Comment:
Does this pass the go race detector?

Reply:
No, there's certainly a race here. If we want to be safe against concurrently closing and writing the checkpoint--which is a decision we can make--we'll need to approach this differently.

Reply:
It's one of those things that's theoretically possible but I doubt any real-world program would ever hit it, since any real program is going to end with a RegisterResourceOutputs on the root resource.

If the go race detector complains we could probably shut it up by doing an atomic load here and atomic stores in the mutator callback, but not sure if we actually care about that.

Reply:
It was easy enough to fix this, so I fixed it.